### PR TITLE
feat: Site

### PR DESCRIPTION
**Pull request by johnletey**
| Light Mode | Dark Mode |
| --- | --- |
| Screenshot 2020-06-10 at 11 42 03 | Screenshot 2020-06-10 at 11 42 08 |
| Screenshot 2020-06-10 at 11 42 16 | Screenshot 2020-06-10 at 11 42 24 |

A new site for this Feather Icons fork. Here are some highlights:

- Pull icon data from the `icons` directory.
  - Site code in main repository.
- Updated UI.
- Simplify site: Use [Next.js](https://nextjs.org/) with TypeScript.

### CODE REVIEW

**Comment by locness3** [comment]
I think we should have a `master` branch and a seperate `site`branch that contains the site.

**Comment by jletey** [comment]
> a seperate `site` branch that contains the site.

@locness3 I disagree with this, as the site is pulling from the `icons` directory, which is very useful for deploy previews of new icons.

However if the site were in a different branch that wouldn't happen.

**Comment by locness3** [comment]
Screen Shot 2020-06-13 at 08 18 09
you might wanna change that

**Comment by jletey** [comment]
@locness3 since the site code isn't deployed to NPM or anything, it's okay to leave this as is.

**Comment by jletey** [comment]
I'm waiting on #4 before I merge this, since that PR allows us to deploy the site successfully.

@locness3 I think the site is okay to deploy, but should I add the attribute customisation feature that [feathericons.com](https://feathericons.com/) has?

**Comment by locness3** [comment]
I don't think so.

Also, we should find a unique name for the project, to avoid confusion

**Comment by jletey** [comment]
> we should find a unique name for the project

Completely agree! That will help when publishing packages etc!

Any ideas 😅 

Also, might be good to transfer this repo to an org as well 🤔

**Comment by locness3** [comment]
> Any ideas 😅

Featherity? (feather + community) 😅

**Comment by jletey** [comment]
> Featherity? (feather + community)

I like it 😅 Created @featherity

**Comment by locness3** [comment]
Didn't expect this name to be actually used :joy:

**Comment by jletey** [comment]
Lol idk I like it it's catchy 😂 We can always rename if we want 😅

**Comment by jletey** [comment]
I have deployed the site, you can now view it at [featherity.netlify.app](https://featherity.netlify.app/).

**Comment by ericfennis** [comment]
Looks awesome!
Is the darkmode gone?

**Comment by jletey** [comment]
@ericfennis There was a small bug, but nothing I can fix on my end 😕 I can add it back if you guys want. I did really love it 😍

**Comment by locness3** [comment]
It seems like the version at https://featherity.netlify.app doesn't have the new icons.

**Comment by locness3** [comment]
Screen Shot 2020-07-01 at 07 36 51
Also, the page is missing a title

**Comment by ericfennis** [comment]
Should we merge this one?
Then the website will automatically be updated when new icons are added to the master branch.

**Comment by locness3** [comment]
> Should we merge this one?

We should first mark it "Ready for review"

**Comment by jletey** [comment]
I'm all for merging this. If anyone has any comments let me know!

File: packages/site/src/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  }
+  useEffect(() => {
+    window.addEventListener("keydown", handleKeyDown);
+    return () => window.removeEventListener("keydown", handleKeyDown);
+  }, []);
+
+  const [numColumns, setNumColumns] = useState(1);
+  useEffect(() => {
+    setNumColumns(Math.floor(window.outerWidth / 160));
+  }, []);
```
Comment (ericfennis) [inline packages/site/src/pages/index.tsx:51]: I noticed that when page loads, all the icons where displayed in full-width for a second. I think we can fix this by calculating and setting it before render.
```suggestion
  const idealColumnCount = () => Math.floor(window.outerWidth / 160);
  const [numColumns, setNumColumns] = useState(idealColumnCount());
  useEffect(() => {
    setNumColumns(idealColumnCount());
  }, []);
```

Comment (jletey) [reply packages/site/src/pages/index.tsx:51]: You can't use the `window` attribute outside of the `useEffect` 😕

Comment (ericfennis) [reply packages/site/src/pages/index.tsx:51]: oh really, didn't know that.
But you can use It in useState right?

```suggestion
  const idealColumnCount = ({outerWidth}) => Math.floor(outerWidth / 160);
  const [numColumns, setNumColumns] = useState(idealColumnCount(window));
  useEffect(() => {
    setNumColumns(idealColumnCount(window));
  }, []);
```

Comment (jletey) [reply packages/site/src/pages/index.tsx:51]: Nope for some reason that doesn't work either 🤔 I don't know why, this is very strange!

**Comment by krisu5** [comment]
Needs Github project link to top-right, imo.

File: packages/site/src/pages/index.tsx (hunk 1 above)
Comment (aelfric) [reply packages/site/src/pages/index.tsx:51]: If you want to get rid of the flash of one large column, I think you can avoid using JS to calculate the column width altogether.  Remove the `useEffect` and `useState` and instead change the grid css to be:

```diff
diff --git a/packages/site/src/pages/index.tsx b/packages/site/src/pages/index.tsx
index c31eb3d..6f3f04a 100644
--- a/packages/site/src/pages/index.tsx
+++ b/packages/site/src/pages/index.tsx
@@ -45,11 +45,6 @@ const IndexPage = ({ data }) => {
     return () => window.removeEventListener("keydown", handleKeyDown);
   }, []);
 
-  const [numColumns, setNumColumns] = useState(1);
-  useEffect(() => {
-    setNumColumns(Math.floor(window.outerWidth / 160));
-  }, []);
-
   return (
     <Layout>
       <Flex direction="column" align="center" justify="center">
@@ -81,7 +76,7 @@ const IndexPage = ({ data }) => {
       </InputGroup>
       {results.length > 0 ? (
         <Grid
-          templateColumns={`repeat(${numColumns > 5 ? 5 : numColumns}, 1fr)`}
+          templateColumns={`repeat(auto-fit, minmax(160px, 1fr))`}
           gap={5}
         >
           {results.map((icon) => {
```

 (sorry I don't know how to use the suggestion feature)

Comment (ericfennis) [reply packages/site/src/pages/index.tsx]: @aelfric Thanks for your code snippet, it fixed the issue!

**Pull request merged**